spyder_vim/vim_widget.py

```diff
@@ -52,10 +52,18 @@ def _move_cursor(self, movement, repeat=1):
         editor.setTextCursor(cursor)
         self._widget.update_vim_cursor()
 
+    def _editor_cursor(self):
+        """returns editor's cursor object"""
+        editor = self._widget.editor()
+        cursor = editor.textCursor()
+        return cursor
+        
     # %% Movement
     def h(self, repeat=1):
-        # TODO: stop at start of line
-        self._move_cursor(QTextCursor.Left, repeat)
+        cursor = self._editor_cursor()
+        if not cursor.atBlockStart():
+            self._move_cursor(QTextCursor.Left)
+            if repeat > 1: self.h(repeat-1)
```

Comment:
Please put the code block in a new line, even if it's small it will be more clear. (same below)